src/probnum/kernels/_kernel.py

```diff
@@ -0,0 +1,302 @@
+            if x1.ndim < 2 and x0.ndim == 2:
+                x1 = np.atleast_2d(x1)
+            if x0.ndim != x1.ndim:  # Shape mismatch
+                raise ValueError(err_msg)
+
+            # Check shapes
+            if (
+                (x0.ndim == 0 and self.input_dim > 1)  # Scalar input
+                or (
+                    x0.ndim == 1  # Vector input
+                    and not (x0.shape[0] == x1.shape[0] == self.input_dim)
+                )
+                or (
+                    x0.ndim == 2  # Matrix input
+                    and not (x0.shape[1] == x1.shape[1] == self.input_dim)
+                )
+            ):
+                raise ValueError(err_msg)
+
+        return x0, x1, equal_inputs
+
+    def _transform_kernelmatrix(
+        self,
+        kerneval: Union[np.ndarray, ScalarArgType],
+        x0_shape: ShapeArgType,
+        x1_shape: ShapeArgType,
+    ) -> Union[np.float_, np.ndarray]:
+        """Transform the kernel matrix based on the given arguments.
+
+        Standardizes the given evaluation of the covariance function to the correct
+        shape determined by the input arguments. If inputs are vectors the
+        output is a numpy scalar if the output dimension of the kernel is 1,
+        otherwise *shape=(output_dim, output_dim)*. If inputs represent multiple
+        observations, then the resulting matrix has *shape=(n0, n1) or
+        (n0, n1, output_dim, output_dim)*.
+
+        Parameters
+        ----------
+        kerneval
+            Covariance function evaluated at ``x0`` and ``x1``.
+        x0_shape :
+            Shape of the first input to the covariance function.
+        x1_shape :
+            Shape of the second input to the covariance function.
+        """
+        if len(x0_shape) <= 1 and len(x1_shape) <= 1:
+            if self.output_dim == 1:
+                return _utils.as_numpy_scalar(kerneval.squeeze())
+            else:
+                kern_shape = ()
+        else:
+            kern_shape = (x0_shape[0], x1_shape[0])
+
+        if self.output_dim > 1:
+            kern_shape += (
+                self.output_dim,
+                self.output_dim,
+            )
+
+        return kerneval.reshape(kern_shape)
+
+    def _as_vectorized_kernel_function(
+        self,
+        fun: Callable[
+            [_InputType, Optional[_InputType]], Union[np.ndarray, ScalarArgType]
+        ],
+    ) -> Callable[[_InputType, Optional[_InputType]], Union[np.ndarray, np.float_]]:
+        """Transform a function into a vectorized covariance function.
+
+        Creates a kernel / covariance function from a (non-vectorized) function
+        :math:`k : \\mathbb{R}^{d_{in}} \\times \\mathbb{R}^{d_{in}} \\rightarrow
+        \\mathbb{R}`. When a kernel matrix is computed the given function ``fun`` is
+        applied to all pairs of inputs.
+
+        Parameters
+        ----------
+        fun
+            (Non-vectorized) covariance function.
+        """
```

Comment:
I do not understand what this function does. (I get the name and the docstring, but not the code)...

Reply:
Function was removed since initialization with a function is no longer possible.